### PR TITLE
expression: support push-down INTDIV to TiKV | tidb-test=pr/2265

### LINKED ISSUE #49050: support push-down INTDIV to TiKV
## Enhancement
It's used by `admin check table`, the performance can be improved if it's supported

### PR DESCRIPTION
**Pull request by wjhuang2016**
### What problem does this PR solve?

Issue Number: close #49050

Problem Summary:
support push-down INTDIV to TiKV
### What changed and how does it work?
Enable it in scalarExprSupportedByTiKV().
### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
support push-down INTDIV to TiKV
```

### CODE REVIEW

**Comment by SeaRise** [comment]
https://github.com/pingcap/tidb/blob/8d181ede017d2ba7d35f72993e6f07b56f6e00de/pkg/expression/expr_to_pb_test.go#L1399-L1412
Maybe a ut could be added here.

**Comment by bb7133** [comment]
> Tests 
>
> - [ ] Unit test
> - [x] Integration test
> - [ ] Manual test (add detailed scripts or steps below)
> - [ ] No need to test

Please correct the test option.

File: pkg/expression/builtin_arithmetic_vec.go [hunk 1]
```diff
@@ -481,7 +469,7 @@ func (b *builtinArithmeticModDecimalSig) vecEvalDecimal(ctx EvalContext, input *
 	y := buf.Decimals()
 	var to types.MyDecimal
 	for i := 0; i < n; i++ {
-		if result.IsNull(i) {
+		if result.IsNull(i) || buf.IsNull(i) {
```
Comment (lcwangchao) [inline pkg/expression/builtin_arithmetic_vec.go:472]: Any bug here before this PR?

File: pkg/expression/builtin_arithmetic.go [hunk 2]
```diff
@@ -787,14 +787,14 @@ func (s *builtinArithmeticIntDivideDecimalSig) Clone() builtinFunc {
...
 func (s *builtinArithmeticIntDivideIntSig) evalInt(ctx EvalContext, row chunk.Row) (int64, bool, error) {
-	b, bIsNull, err := s.args[1].EvalInt(ctx, row)
-	if bIsNull || err != nil {
-		return 0, bIsNull, err
-	}
 	a, aIsNull, err := s.args[0].EvalInt(ctx, row)
 	if aIsNull || err != nil {
 		return 0, aIsNull, err
 	}
+	b, bIsNull, err := s.args[1].EvalInt(ctx, row)
```
[the remark this answers is not in the document]
Comment (wjhuang2016) [reply pkg/expression/builtin_arithmetic.go:794]: To be compatible with the vec-execution version and MySQL. It affects the 'shortcut evaluation` when it meets an error.

File: pkg/expression/builtin_arithmetic_vec.go (hunk 1 above)
Comment (wjhuang2016) [reply pkg/expression/builtin_arithmetic_vec.go:472]: No, but the previous code is strange. I make it correct.

**Review by lcwangchao — APPROVED** [review]

**Review by breezewish — APPROVED** [review]

**Review by XuHuaiyu — APPROVED** [review]

**Pull request merged**